packages/seedless-onboarding-controller/src/SeedlessOnboardingController.ts

```diff
@@ -0,0 +1,220 @@
+import type {
+  ControllerGetStateAction,
+  RestrictedMessenger,
+  StateMetadata,
+} from '@metamask/base-controller';
+import { BaseController } from '@metamask/base-controller';
+import {
+  encryptWithKey,
+  decryptWithKey,
+  keyFromPassword,
+  generateSalt,
+  importKey,
+  exportKey,
+} from '@metamask/browser-passworder';
+import type {
+  KeyringControllerState,
+  KeyringControllerStateChangeEvent,
+} from '@metamask/keyring-controller';
+
+import { ToprfAuthClient } from './ToprfClient';
+import type {
+  AuthenticateUserParams,
+  CreateSeedlessBackupParams,
+  Encryptor,
+  NodeAuthTokens,
+} from './types';
+
+const controllerName = 'SeedlessOnboardingController';
+
+// State
+export type SeedlessOnboardingControllerState = {
+  /**
+   * The node auth tokens from OAuth User authentication after the Social login.
+   *
+   * This values are used to authenticate users when they go through the Seedless Onboarding flow.
+   */
+  nodeAuthTokens?: NodeAuthTokens;
+  /**
+   * Indicates whether the user has already fully/partially completed the Seedless Onboarding flow.
+   *
+   * An encryption key is generated from user entered password using Threshold OPRF and the seed phrase is encrypted with the key.
+   * During the Seedless Onboarding Authentication step, TOPRF services check whether user has already generated the encryption key.
+   *
+   * If this value is `true`, we can assume that user already has completed the `SeedPhrase` generation step, and user will have to
+   * fetch the `SeedPhrase` with correct password. Otherwise, users will be asked to set up seedphrase and password, first.
+   */
+  hasValidEncryptionKey?: boolean;
+};
+
+// Actions
+export type SeedlessOnboardingControllerGetStateActions =
+  ControllerGetStateAction<
+    typeof controllerName,
+    SeedlessOnboardingControllerState
+  >;
+
+export type AllowedEvents = KeyringControllerStateChangeEvent;
+export type AllowedActions = SeedlessOnboardingControllerGetStateActions;
+
+export const defaultState: SeedlessOnboardingControllerState = {};
+const seedlessOnboardingMetadata: StateMetadata<SeedlessOnboardingControllerState> =
+  {
+    nodeAuthTokens: {
```

Comment:
Should allow the seedless onboarding controller's constructor to set the initial state
export getDefaultSeedlessOnboardingControllerState that return this default state